synapse/lib/agenda.py

```diff
@@ -809,7 +809,20 @@ async def _scheduleLoop(self):
                     logger.warning(mesg,
                                    extra={'synapse': {'iden': appt.iden, 'name': appt.name}})
                 else:
-                    await self._execute(appt)
+                    try:
+                        await self._execute(appt)
+                    except Exception as e:
+                        extra = {'iden': appt.iden, 'name': appt.name, 'user': appt.creator, 'view': appt.view}
+                        user = self.core.auth.user(appt.creator)
+                        if user is not None:
+                            extra['username'] = user.name
+                        if isinstance(e, s_exc.SynErr):
+                            mesg = e.get('mesg', str(e))
+                        else:  # pragma: no cover
+                            mesg = str(e)
+                        logger.exception(f'Agenda error running appointment {appt.iden} {appt.name}: {mesg}',
```

Comment:
Last I checked, general guidance is to not use f-strings with logging functions. I think it's because the string substitutions happen every time with f-strings where if you use the format specifiers, the substitutions only happen if the message actually gets logged. 

```suggestion
                        logger.exception('Agenda error running appointment %s %s: %s', appt.iden, appt.name, mesg,
```

Oh hey, there's even vtx guidance saying the same thing: https://github.com/vertexproject/synapse/blob/master/docs/synapse/contributing/contributing.rst#L234. Although it looks like there's violations of this everywhere. So ... *shrug*

Reply:
Good observation - that guidance predates the availability of f-strings in the codebase at all.

Generally we don't have logging occurring inside of performance-sensitive areas of code, and [ I find ] the f-string syntax is much easier to read / write. The fstring performance is also better than `%` formatting, so in the event of actually recording the log event, we're coming out ahead there.